GitUI/UserControls/RevisionGrid/RevisionGridControl.cs

```diff
@@ -873,9 +873,15 @@ public void PerformRefreshRevisions(Func<RefsFilter, IReadOnlyList<IGitRef>> get
             // The information from step 1. is therefore protected with a semaphore.
             // This local semaphore instance is bound in closures below.
             System.Threading.SemaphoreSlim semaphoreCurrentCommit = new(initialCount: 0, maxCount: 1);
-            _isRefreshingRevisions = true;
 
             // Protected by semaphoreCurrentCommit
+
+            _isRefreshingRevisions = true;
+            //// TODO: What if we do not reach the end of OnRevisionReadCompleted(), where _isRefreshingRevisions is reset?!?
```

Comment:
Can you think of a scenario?

Reply:
`RevisionReader.ExecuteAsync` does call `OnCompleted` only if it reaches the end of the function (and no cancellation was requested).
So if an exception is thrown, `_isRefreshingRevisions` remains `true` and blocks any further updates.
Most likely, this hat not bitten us because this function mainly starts further async operations. Exceptions from these are forwarded to the UI thread and do not interrupt `ExecuteAsync` itself.

Reply:
_isRefreshingRevisions is set to false also in OnError now, so it is possible to reload without restarting GE now